iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/WeakHashAlgorithmsCheck.java

```diff
@@ -0,0 +1,151 @@
+
+  private static final String MESSAGE = "Using weak hashing algorithms is security-sensitive.";
+  private static final Set<String> OPENSSL_SENSITIVE_SUBCOMMAND = Set.of("md5", "sha1", "rmd160", "ripemd160");
+  private static final Set<String> OPENSSL_SENSITIVE_DGST_OPTION  = Set.of("-md2", "-md4", "-md5", "-sha1", "-ripemd160", "-ripemd", "-rmd160");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, WeakHashAlgorithmsCheck::checkRun);
+  }
+
+  private static void checkRun(CheckContext ctx, RunInstruction runInstruction) {
+    CommandDetector sensitiveOpenSslSubcommand = CommandDetector.builder()
+      .with("openssl"::equals)
+      .with(OPENSSL_SENSITIVE_SUBCOMMAND::contains)
+      .build();
+    CommandDetector sensitiveOpenSslDgst = CommandDetector.builder()
+      .with("openssl"::equals)
+      .with("dgst"::equals)
+      .withOptional(s -> !OPENSSL_SENSITIVE_DGST_OPTION.contains(s) && s.startsWith("-"))
+      .with(OPENSSL_SENSITIVE_DGST_OPTION::contains)
+      .build();
+
+    sensitiveOpenSslSubcommand.search(runInstruction.arguments())
+      .forEach(sensitiveOpensslCall -> ctx.reportIssue(TextRanges.mergeRanges(sensitiveOpensslCall), MESSAGE));
+    sensitiveOpenSslDgst.search(runInstruction.arguments())
+      .forEach(sensitiveOpensslCall -> ctx.reportIssue(TextRanges.mergeRanges(sensitiveOpensslCall), MESSAGE));
+  }
+
+  private static class CommandDetector {
+
+    List<Predicate<String>> predicates;
+    List<Boolean> isOptionalPredicates;
+
+    private CommandDetector(List<Predicate<String>> predicates, List<Boolean> isOptionalPredicates) {
+      this.predicates = predicates;
+      this.isOptionalPredicates = isOptionalPredicates;
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    /**
+     * Return all block of arguments which match the list of predicates.
+     */
+    private List<List<Argument>> search(List<Argument> arguments) {
+      List<ArgumentResolution> resolvedArgument = arguments.stream().map(ArgumentResolution::of).collect(Collectors.toList());
+      List<List<Argument>> argumentBlockIssue = new ArrayList<>();
+
+      int argIndex = 0;
+      while (argIndex < arguments.size()) {
+        Integer sizeMatch = fullMatch(resolvedArgument, argIndex);
+        if (sizeMatch != null) {
+          argumentBlockIssue.add(arguments.subList(argIndex, argIndex + sizeMatch));
+          argIndex += sizeMatch;
+        } else {
+          argIndex++;
+        }
+      }
+      return argumentBlockIssue;
+    }
+
+    /**
+     * If the provided list of resolved argument match with the list of predicates, return the size of matched predicates, ignoring optional predicates that didn't match.
+     * Otherwise, it will return null.
+     */
+    @CheckForNull
+    private Integer fullMatch(List<ArgumentResolution> resolvedArgument, int argIndex) {
+      int nbMatched = 0;
```

Comment:
What does `nb` means?

Reply:
Old french habbit ^^' Mean something like 'nombre', which could translate to 'quantity', I will change to a better name.